atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTableTest.java

```diff
@@ -49,57 +48,67 @@
 
     @Test
-    public void shouldCompactIfVacuumWasPerformedBeforeCompactInterval() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(new Timestamp(10), null, new Timestamp(9), null);
-        assertTrue(postgresDdlTable.checkIfTableHasNotBeenCompactedForCompactInterval());
+    public void shouldCompactIfVacuumWasNeverPerformedAndTheDbTimeIsMoreThanCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(null, COMPACT_INTERVAL_MILLIS * SMALL_POSITIVE_FACTOR);
 
-        assertThatVacuumWasPerformed(sqlConnection);
+        assertThatVacuumWasPerformed(sqlConnection, true);
     }
 
     @Test
-    public void shouldCompactIfAutoVacuumWasPerformedBeforeCompactInterval() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(null, new Timestamp(10), null, new Timestamp(9));
-        assertTrue(postgresDdlTable.checkIfTableHasNotBeenCompactedForCompactInterval());
+    public void shouldCompactIfVacuumWasPerformedExactlyBeforeCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS, NOW_MILLIS);
 
-        assertThatVacuumWasPerformed(sqlConnection);
+        assertThatVacuumWasPerformed(sqlConnection, true);
     }
 
+    @Test
+    public void shouldCompactIfVacuumWasPerformedBeforeCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS * SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
+
+        assertThatVacuumWasPerformed(sqlConnection, true);
+    }
 
     @Test
-    public void shouldNotCompactIfAutoVacuumWasPerformedWithinCompactInterval() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(null, new Timestamp(95), null, new Timestamp(92));
+    public void shouldNotCompactIfVacuumWasPerformedWithinCompactInterval() {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
         assertThatVacuumWasNotPerformed(sqlConnection);
     }
 
-
     @Test
-    public void shouldNotCompactIfVacuumWasPerformedWithinCompactInterval() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(new Timestamp(95), null, new Timestamp(92), null);
+    public void shouldNotCompactIfVacuumTimestampExceedsNowTimestampByLessThanCompactInterval() {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS + COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
         assertThatVacuumWasNotPerformed(sqlConnection);
     }
 
     @Test
-    public void shouldNotCompactIfVacuumDidOccurInLastCompactMillis() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(new Timestamp(5), null, new Timestamp(90), new Timestamp(10));
-
+    public void shouldNotCompactIfVacuumTimestampExceedsNowTimestampByMoreThanCompactInterval() {
```

Comment:
I'm worried about this case, because it means we have clock drift. In a bad case of this, we might be virtually permanently be disabling vacuum :/

Maybe we should have `getMillisSinceLastCompact()` return `abs(last - current)` to guard against this? Or at least log something at WARN+.

Reply:
Yup, this is a legit concern. The proposed fix sounds reasonable.

Reply:
Tried to fix it this way but a clock skew could be so that the vacuum timestamp remains in the compact interval always. Also, Postgres timestamps are used all around so a lot more things will be bad in the read-write paths than just disabling vacuums, lets not have this to keep it cleaner. DBAs should be taking care of there being no clock skews.